src/json.hpp.re2c

```diff
@@ -618,7 +700,8 @@ class basic_json
         boolean,        ///< boolean value
         number_integer, ///< number value (integer)
         number_float,   ///< number value (floating-point)
-        discarded       ///< discarded by the the parser callback function
+        discarded,      ///< discarded by the the parser callback function
+        number_unsigned ///< number value (unsigned integer)
```

Comment:
I put it at the end because I was concerned that renumbering the enum might break existing code that was expecting certain types to have certain numerical values.  I agree that after number_integer would be a much better position aesthetically. If you are happy that the renumbering won't cause any flow on problems I'll change it.  The added benefit is that it removes the necessity to insert a filler value in the vector that is used for the type ordering comparison (is null < bool etc).

The Travis problem is caused by Valgrind - I'm not sure why but it also happens when I run the unit tests locally under Valgrind.  For some reason Valgrind is truncating the parsed value.  Seems rather bizarre but I'm looking into it.